python/seldon_core/metrics.py

```diff
@@ -1,10 +1,115 @@
+from prometheus_client.core import (
+    HistogramMetricFamily,
+    GaugeMetricFamily,
+    CounterMetricFamily,
+    CollectorRegistry,
+)
+from prometheus_client import exposition
+from prometheus_client.utils import floatToGoString
+
+from multiprocessing import Manager
+
+import numpy as np
+
 from typing import List, Dict
+import logging
+import os
+
+
+logger = logging.getLogger(__name__)
+
 
 COUNTER = "COUNTER"
 GAUGE = "GAUGE"
 TIMER = "TIMER"
 
 
+def generate_metrics(metrics):
```

Comment:
Given that this function seems to always require the metrics, it seems that it would make more sense for it to be part of the SeldonMetrics class - you could move the myregistry.register(...) into the __init__ and store the myregistry as a class variable - then you can expose the generate latest through a class function

Reply:
Moving this to be a method of `SeldonMetrics` sounds good.

However, I would avoid storing a single `CollectorRegistry()` instance between threads / processes for same reason I am dynamically creating `GaugeMetricFamily`, `CounterMetricFamily` and `HistogramMetricFamily` whenever I need to collect metrics.
Not saying I am not sure if it would even work as it contains `threading.Lock`.

The main purpose of `SeldonMetrics` is to store state of metrics as basic Python type inside process-safe dictionary.

Reply:
`generate_metrics` is now method of SeldonMetrics`